querycheck/query_check.go

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package querycheck
+
+import (
+	"context"
+
+	tfjson "github.com/hashicorp/terraform-json"
+)
+
+// QueryResultCheck defines an interface for implementing test logic that checks a query file and then returns an error
+// if the query file does not match what is expected.
+type QueryResultCheck interface {
+	// CheckQuery should perform the query check.
+	CheckQuery(context.Context, CheckQueryRequest, *CheckQueryResponse)
+}
+
+// CheckQueryRequest is a request for an invoke of the CheckQuery function.
+type CheckQueryRequest struct {
+	// Query represents a parsed query file, retrieved via the `terraform show -json` command.
+	Query *[]tfjson.ListResourceFoundData
```

Comment:
nit: Also I don't think we need this to be a pointer to a slice here. Slices are already by reference under the hood